### DISCUSSION TITLE
Record file duration increase with H265 Smart Codec on Dahua cams

### DISCUSSION THREAD

**Discussion by Al-Dek** [blakeblackshear/frigate#12384] [Ask A Question] [answered]
### What is your question:

I have 4 Dahua cams (DH-IPC-HFW2449S-S-IL) installed in Frigate.
Everything works perfect, but i noticed one thing - if I switch on Smart Codec for H265 record file duration increases from 10 to 12 seconds. (and vice versa :-) )
Settings:
Resolution: 2688x1520 
Encode mode: H265
Smart codec: ON
FPS: 16
Bit Rate Type: VBR
Max Bit Rate: 1536

Why, what is the reason ?

**Comment by Al-Dek** [comment]
Update:
if fps=12, the duration of the recording file increases to 16 seconds...
?

**Comment by NickM-27** [comment, marked as answer]
Because a segment has to begin and end on a key frame, smart most make it send those less often
